### PR TITLE
Refactor Load and Performance Doc

### LINKED ISSUE #1851: Load and Performance Testing Doc Update
Re: [Load and Performance Testing](https://pantheon.io/docs/load-and-performance-testing)
## Issue Description:

Title of doc is Load and Performance Testing, but doc intro is all about Load testing, so let’s revise it!

We should talk about 
1. **Load** vs **performance** testing, highlight value of smooth scaling and consistent infrastructure. Performance characteristics of Live are same as other environments so if it’s slow on Dev, don’t deploy Live. If you uncover performance issues with a single request, no need to conduct a load test, fix the performance issue first!
2. How to do performance testing in Dev/Test. Highlight New Relic
3. Load testing in Live environment **before launch**
4. Use math to calculate load capacity **after launch**. After launch, you can establish a baseline that X response time will let you handle Y traffic. If X degrades in Dev/Test, that will impact how much traffic Live can handle. We do not recommend load testing in Live after launch because you risk overwhelming your Live site and causing downtime. 

Add heading:
How to calculate how much traffic you can handle:

If you know traffic will increase,  you can understand the relationship between your site’s performance and how much load it can handle by doing a little math.

Maybe best way to communicate above is via a table, like this interactive spreadsheet? (https://docs.google.com/a/getpantheon.com/spreadsheets/d/1K9BdmdJDqHMvHMaerZZmCcflSvmEJhgBsGL8zwjpxE8/edit?usp=sharing)

For example, if your application's average response time is Xms and the Live environment gets Y requests per minute (rpm), then a Business plan with 16 php workers will handle the load. If you anticipate a traffic increase and you’d like to upgrade to Elite, then <contact Sales>.   (validate this math based on new pricing comparison doc)

Note that
Dev environments defaults to not cache static assets like CSS and JS to facilitate development. This will affect frontend performance of those environments, and you should be aware of that if you are perf testing there.

Also include link to https://pantheon.io/docs/guides/load-testing-with-blazemeter

### PR DESCRIPTION
**Pull request by rachelwhitton**
Closes #1851 
Replaces #2109 
## Effect
PR includes the following changes:
- Rewrite of performance test documentation

Todo: 
- [ ] Peer review from another EOM (@aeligature?) and/or @ari-gold 
- [ ] Additional copy review from @alexfornuto following peer review

### CODE REVIEW

File: source/_docs/load-and-performance-testing.md [hunk 1]
```diff
@@ -1,200 +1,73 @@
...
+  * **Load Tests**: Determine how many concurrent users the site is expected to serve based on historical analytics for the site. Identify the peak hourly sessions and average session duration, then do some math: `hourly_sessions / (60 / average_duration) = Concurrent Users`
+
+
+###Running the Tests
+If this is a performance test, be sure to run the test on a development environment (Dev or [Multidev](/docs/multidev)) without caching. Run load tests on the Live environment before launching the site. If the site is already launched, use the Test environment instead.
+<div markdown="1" class="alert alert-danger" role="alert">
+###Warning {.info}
+We do not recommend load testing on the Live environment if the site has already launched because you risk overwhelming your live site and causing downtime.
+</div>
+Note the start time for the test. As the test executes, it's a good idea to keep a close eye on [log files](/docs/logs). Make note of any errors and warnings that pop up during test to fix.
```
Comment (rachelwhitton) [inline source/_docs/load-and-performance-testing.md]: Which log files should I watch? Do they differ based on the type of test being run?

File: source/_docs/load-and-performance-testing.md [hunk 2]
```diff
@@ -1,200 +1,73 @@
...
+  * [Locust](http://locust.io/)
+
+  The Pantheon onboarding team uses Locust, an open source load testing tool, because it's fairly easy to build out test scripts and it allows you to crawl the site instead of using predefined URLs. Crawling the site has the added benefit of hitting every page that is linked to anywhere on the site.  This exposes edge case performance bottlenecks that would have gone undetected under more rigid tests.
+
+  Ultimately, is doesn't matter what tool you decide to use as long as it allows you to test your site properly.  Be sure to allow for any authenticated traffic as well as anonymous.  
+
+3. Determine how much load to apply for your test.
+
+  * **Performance Tests**: Smaller loads should suffice, as you should be able to see transactional bottlenecks with 10-20 concurrent users.
+  * **Load Tests**: Determine how many concurrent users the site is expected to serve based on historical analytics for the site. Identify the peak hourly sessions and average session duration, then do some math: `hourly_sessions / (60 / average_duration) = Concurrent Users`
```
Comment (rachelwhitton) [inline source/_docs/load-and-performance-testing.md]: How do I determine load to apply in the test after calculating concurrent users?

File: source/_docs/load-and-performance-testing.md [hunk 3]
```diff
@@ -1,200 +1,75 @@
...
+ * SaaS Solutions
+  * [Blazemeter](https://www.blazemeter.com)
+  * [Load Impact](https://loadimpact.com)
+ * Open Source tools
+  * [Jmeter](http://jmeter.apache.org)
+  * [Locust](http://locust.io/)
+
+  The Pantheon onboarding team uses Locust, an open source load testing tool. Locust makes it easy to build out test scripts, and it allows you to crawl the site instead of using predefined URLs. Crawling the site has the added benefit of loading every page that is linked to anywhere on the site. This exposes edge case performance bottlenecks that would have gone undetected under tests with predifined URLs.
+
+  Ultimately, it doesn't matter what tool you use as long as you to test your site properly. Be sure to allow for any authenticated traffic as well as anonymous.
```
Comment (ari-gold) [inline source/_docs/load-and-performance-testing.md]: "Be sure to allow for any authenticated traffic as well as anonymous" - Not sure we should just assert this in passing. Load testing authenticated users can be difficult.

File: source/_docs/load-and-performance-testing.md [hunk 4]
```diff
@@ -1,200 +1,75 @@
...
+  * [Jmeter](http://jmeter.apache.org)
+  * [Locust](http://locust.io/)
+
+  The Pantheon onboarding team uses Locust, an open source load testing tool. Locust makes it easy to build out test scripts, and it allows you to crawl the site instead of using predefined URLs. Crawling the site has the added benefit of loading every page that is linked to anywhere on the site. This exposes edge case performance bottlenecks that would have gone undetected under tests with predifined URLs.
+
+  Ultimately, it doesn't matter what tool you use as long as you to test your site properly. Be sure to allow for any authenticated traffic as well as anonymous.  
+
+3. Determine how much load to apply.
+
+  * **Performance Tests**: Smaller loads should suffice, as you should be able to see transactional bottlenecks with 10-20 concurrent users.
```
Comment (ari-gold) [inline source/_docs/load-and-performance-testing.md]: Why 10-20? A single request can give you all you need, no?
We should explain how to use tools like Google Dev Tools for website performance optimization or at least link to resources like:
 https://hpbn.co/ 
https://www.udacity.com/course/website-performance-optimization--ud884

File: source/_docs/load-and-performance-testing.md [hunk 5]
```diff
@@ -1,200 +1,75 @@
...
-## Performance vs. Scalability
-
-There are two things to test for:
-
-1. **Performance**: the response time for an individual request
-2. **Scalability**: the ability to deliver with optimal response time to a larger number of concurrent requests
-
-High-performance is the ability to deliver a page in under a second; scalability is the ability to deliver that page in under a second for many requests. It's important to understand the difference between these two dimensions and that there are trade-offs between performance and scalability.
-
-## Verify Varnish is Working
```
Comment (ari-gold) [inline source/_docs/load-and-performance-testing.md]: Seems like verifying Varnish is working is still important before doing a load test? Maybe this can be more concise?

**Comment by rachelwhitton** [comment]
We're going to deploy this as an iterative improvement and circle back to address suggestions not implemented here. See #2251 to track

File: source/_docs/load-and-performance-testing.md [hunk 6]
```diff
@@ -1,200 +1,75 @@
...
-			<td><a href="https://www.blitz.io/pricing#/subscriptions">Pricing</a></td>
-		</tr>
-	</tbody>
-</table>
+Load and performance tests are critical steps in going live procedures, as they help expose and identify potential performance killers. These tests provide insight for how a site will perform in the wild under peak traffic spikes.
+
+## Load vs Performance Testing
+Before you start, it's important to understand the difference between load and performance testing and know when to use each.
+### Performance Testing
+Performance testing is the process in which you measure an application's response time to proactively expose bottlenecks. These tests should be regularly executed as part of routine maintenance. Additionally, you should run these test before any load testing. If your application is not performing well, then you can be assured that the load test will not go well.
```
Comment (obicke) [inline source/_docs/load-and-performance-testing.md]: In general, I'd favor suggesting that clients: 

- "**refer to New Relic reports regularly** to identify improvements or degradation of performance
- "**perform performance test occasionally** to proactively exposed potential bottlenecks and to identify opportunities for optimization" and to 
- perform load tests in advance of anticipated major-traffic events, or prior to launching sites after major overhauls, remembering to provide enough time to fix any issues identified".

File: source/_docs/load-and-performance-testing.md [hunk 7]
```diff
@@ -1,200 +1,75 @@
...
-	</tbody>
-</table>
+Load and performance tests are critical steps in going live procedures, as they help expose and identify potential performance killers. These tests provide insight for how a site will perform in the wild under peak traffic spikes.
+
+## Load vs Performance Testing
+Before you start, it's important to understand the difference between load and performance testing and know when to use each.
+### Performance Testing
+Performance testing is the process in which you measure an application's response time to proactively expose bottlenecks. These tests should be regularly executed as part of routine maintenance. Additionally, you should run these test before any load testing. If your application is not performing well, then you can be assured that the load test will not go well.  
+
+The scope of performance tests should be limited to the application itself on a development environment (Dev or [Multidev](/docs/multidev)) without caching. This will give you an honest look into your application and show exactly how uncached requests will perform. You can bypass cache by [setting the `no-cache` HTTP headers](/docs/cache-control) in responses.
```
Comment (obicke) [inline source/_docs/load-and-performance-testing.md]: My understanding is that the dev environment has a default time-to-live of zero for dev, which implies no caching, but that things that Pantheon Advanced Page Cache may override this to be non-zero value. While a no-cache header may help, this may depend on when this get executed. Suggesting to disable caching via the UI is an option, with an emphasis to remember to re-enable prior to pushing to prod.

File: source/_docs/load-and-performance-testing.md [hunk 8]
```diff
@@ -1,200 +1,75 @@
...
+
+### Load Testing
+Load testing is the process in which you apply requests to your site that will represent the most load that your site will face once it is live.  This test will ensure that the site can withstand the peak traffic spikes after launch. This test should be done on the Live environment before the site has launched, after performance testing.
+
+If your site is already live, then you should run load tests on the Test environment. Keep in mind that the Test environment has one application container, while Live environments on sites with a service level of Business and above can have multiple application containers serving the site. So try to run a proportionate amount of traffic based on how many containers you currently have on your Live environment.
```
Comment (obicke) [inline source/_docs/load-and-performance-testing.md]: The EOM team is the best source for the algorithm we use.

File: source/_docs/load-and-performance-testing.md [hunk 9]
```diff
@@ -1,200 +1,75 @@
...
+2. Select a load testing tool:
+
+ * SaaS Solutions
+  * [Blazemeter](https://www.blazemeter.com)
+  * [Load Impact](https://loadimpact.com)
+ * Open Source tools
+  * [Jmeter](http://jmeter.apache.org)
+  * [Locust](http://locust.io/)
+
+  The Pantheon onboarding team uses Locust, an open source load testing tool. Locust makes it easy to build out test scripts, and it allows you to crawl the site instead of using predefined URLs. Crawling the site has the added benefit of loading every page that is linked to anywhere on the site. This exposes edge case performance bottlenecks that would have gone undetected under tests with predifined URLs.
```
Comment (obicke) [inline source/_docs/load-and-performance-testing.md]: The EOM team should be asked to update this section.

File: source/_docs/load-and-performance-testing.md (hunk 3 above)
Comment (obicke) [inline source/_docs/load-and-performance-testing.md]: I agree that authenticated user testing is a complex task and thus the generic statement should be along the lines of "It is important for Load Testing to test against the anticipated traffic patterns of the site, both in terms of traffic volume and authenticated/anonymous proportion. Note that testing authenticated workflows is considerably more complex requiring more time, skills and iterations."

File: source/_docs/load-and-performance-testing.md (hunk 4 above)
Comment (obicke) [inline source/_docs/load-and-performance-testing.md]: IMO, you want to generate more than single request to tease out potential bottlenecks.

Also, I know that we have a Quicksilver example that will use free loader.io account to automatically run this level of test on each push to Test environment. Not only does this result in automated testing procedures, it provides a standard profile that you can see in New Relic. Here's a related link, but we need better: https://github.com/pantheon-systems/quicksilver-examples/issues/110

File: source/_docs/load-and-performance-testing.md (hunk 5 above)
Comment (obicke) [reply source/_docs/load-and-performance-testing.md]: Is this still the case now that Global CDN is in place?

File: source/_docs/load-and-performance-testing.md (hunk 3 above)
Comment (obicke) [reply source/_docs/load-and-performance-testing.md]: I edited

File: source/_docs/load-and-performance-testing.md (hunk 4 above)
Comment (obicke) [reply source/_docs/load-and-performance-testing.md]: IMO, this is good to go (i.e. no edits needed). A separate issue should be created, if/when we want to include reference to the loader.io Quicksilver example.

File: source/_docs/load-and-performance-testing.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (obicke) [reply source/_docs/load-and-performance-testing.md]: Added some of these notions.

File: source/_docs/load-and-performance-testing.md [hunk 10]
```diff
@@ -1,200 +1,75 @@
...
+
+  Ultimately, it doesn't matter what tool you use as long as you to test your site properly. Be sure to allow for any authenticated traffic as well as anonymous.  
+
+3. Determine how much load to apply.
+
+  * **Performance Tests**: Smaller loads should suffice, as you should be able to see transactional bottlenecks with 10-20 concurrent users.
+  * **Load Tests**: Determine how many concurrent users the site is expected to serve based on historical analytics for the site. Identify the peak hourly sessions and average session duration, then do some math: `hourly_sessions / (60 / average_duration) = Concurrent Users`
```
[the remark this answers is not in the document]
Comment (obicke) [reply source/_docs/load-and-performance-testing.md]: Load tests should not be run on Test, rather performance test can/should be run there. In terms of providing formulas, it is complicated by the fact that to run "proportionate amount of traffic" on Test involves knowing the number of appservers on Live, which clients can't determine on their own (other than asking Support, or looking at New Relic, which will include decommissioned appservers for some time).
